agent/agent_endpoint_test.go

```diff
@@ -6834,33 +6834,26 @@ func TestAgentConnectCALeafCert_good(t *testing.T) {
 			// Set a new CA
 			ca3 := connect.TestCAConfigSet(t, a, nil)
 
-			resp := httptest.NewRecorder()
 			req, err := http.NewRequest("GET", "/v1/agent/connect/ca/leaf/test", nil)
 			require.NoError(t, err)
-			obj, err = a.srv.AgentConnectCALeafCert(resp, req)
-			require.NoError(t, err)
-			issued2 := obj.(*structs.IssuedCert)
-			require.NotEqual(t, issued.CertPEM, issued2.CertPEM)
-			require.NotEqual(t, issued.PrivateKeyPEM, issued2.PrivateKeyPEM)
 
-			// Verify that the cert is signed by the new CA
-			requireLeafValidUnderCA(t, issued2, ca3)
+			retry.Run(t, func(r *retry.R) {
+				resp := httptest.NewRecorder()
+				a.srv.h.ServeHTTP(resp, req)
 
-			// Should not be a cache hit!
-			require.Equal(t, "MISS", resp.Header().Get("X-Cache"))
-		}
+				// Should not be a cache hit!
+				require.Equal(r, "MISS", resp.Header().Get("X-Cache"))
```

Comment:
I guess it's expected that we have few cache hits before updating, shouldn't we make this in its own retry loop to increase the change of the test passing?

Reply:
I think something about the design of leaf cache or the test makes it impossible to get a cache HIT. I tried making a subtest for hitting cache and never got it to pass.

Reply:
There's a test below called `TestAgentConnectCALeafCert_nonBlockingQuery_after_blockingQuery_shouldNotBlock` which does assert for a cache hit; I didn't want to spend too much time looking at this 😆

Reply:
No worries, it was just an idea. Let's see if it's still failing before trying to over optimize 😄